spring-cloud-stream/src/test/java/org/springframework/cloud/stream/function/SourceToFunctionsSupportTests.java

```diff
@@ -0,0 +1,226 @@
+						TestChannelBinderConfiguration.getCompleteConfiguration(SupplierConfiguration.class))
+						.web(WebApplicationType.NONE)
+						.run("--spring.cloud.stream.function.name=number|concatWithSelf|mmultiplyByTwo",
+								"--spring.jmx.enabled=false");
+
+		OutputDestination target = context.getBean(OutputDestination.class);
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("22".getBytes(StandardCharsets.UTF_8));
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("44".getBytes(StandardCharsets.UTF_8));
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("66".getBytes(StandardCharsets.UTF_8));
+		//etc
+	}
+
+	@EnableAutoConfiguration
+	@Import(ProvidedMessageSourceConfiguration.class)
+	public static class SupplierConfiguration {
+		AtomicInteger counter = new AtomicInteger();
+		@Bean
+		public Supplier<String> number() {
+			return () -> String.valueOf(counter.incrementAndGet());
+		}
+
+		@Bean
+		public Function<String, String> concatWithSelf() {
+			return x -> x + x;
+		}
+
+		@Bean
+		public Function<Flux<String>, Flux<String>> mmultiplyByTwo() {
+			return x -> x.map(i -> String.valueOf(Integer.valueOf(i) * 2));
+		}
+	}
+
+	@EnableAutoConfiguration
+	@Import(ExistingMessageSourceConfiguration.class)
+	public static class FunctionsConfiguration {
+
+		@Bean
+		public Function<String, String> toUpperCase() {
+			return String::toUpperCase;
+		}
+
+		@Bean
+		public Function<String, String> concatWithSelf() {
+			return x -> x + ":" + x;
+		}
+	}
+
+	/**
+	 * This configuration essentially emulates our existing app-starters for Sources
+	 * and essentially demonstrates how a function(s) could be applied to an existing
+	 * source via {@link IntegrationFlowFunctionSupport} class.
+	 */
+	@EnableBinding(Source.class)
+	public static class ExistingMessageSourceConfiguration {
+
+		@Autowired
+		private Source source;
+
+		@Bean
+		public IntegrationFlow messageSourceFlow(IntegrationFlowFunctionSupport functionSupport) {
+			Supplier<Message<String>> messageSource = () -> MessageBuilder.withPayload("hello function")
+					.setHeader(MessageHeaders.CONTENT_TYPE, MimeTypeUtils.TEXT_PLAIN).build();
+
+			IntegrationFlowBuilder flowBuilder = functionSupport.integrationFlowFromProvidedSupplier(messageSource);
+
+			if (!functionSupport.andThenFunction(flowBuilder, source.output())) {
+				flowBuilder = flowBuilder.channel(source.output());
+			}
+
+			return flowBuilder.get();
+		}
+
+	}
+
+	@EnableBinding(Source.class)
+	public static class ProvidedMessageSourceConfiguration {
+
+		@Autowired
+		private Source source;
+
+		@Autowired
+		private FunctionProperties functionProperties;
+
+		@Bean
+		public IntegrationFlow messageSourceFlow(IntegrationFlowFunctionSupport functionSupport) {
+			Assert.hasText(functionProperties.getName(), "Supplier name must be provided");
+
+			IntegrationFlowBuilder flowBuilder = functionSupport
+					.integrationFlowFromNamedSupplier()
+					.split()
```

Comment:
I thought you were going to have this `.split()` as out-of-the-box feature.
Or do you leave a `Flux` processing to the end-user?